test/functional/rpc_psbt.py

```diff
@@ -230,6 +230,47 @@ def test_sighash_mismatch(self):
 
         wallet.unloadwallet()
 
+    def test_sighash_adding(self):
+        self.log.info("Test adding of sighash type field")
+        self.nodes[0].createwallet("sighash_adding")
+        wallet = self.nodes[0].get_wallet_rpc("sighash_adding")
+        def_wallet = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+
+        outputs = [{wallet.getnewaddress(address_type="bech32"): 1}]
+        if self.options.descriptors:
+            outputs.append({wallet.getnewaddress(address_type="bech32m"): 1})
+        def_wallet.send(outputs)
+        self.generate(self.nodes[0], 6)
+
+        # Make a PSBT
+        psbt = wallet.walletcreatefundedpsbt(wallet.listunspent(), [{def_wallet.getnewaddress(): 0.5}])["psbt"]
+        psbt = wallet.walletprocesspsbt(psbt=psbt, sighashtype="ALL|ANYONECANPAY", finalize=False)["psbt"]
+
+        # Check that the PSBT has a sighash field on all inputs
+        dec_psbt = self.nodes[0].decodepsbt(psbt)
+        for input in dec_psbt["inputs"]:
+            assert_equal(input["sighash"], "ALL|ANYONECANPAY")
+
+        # Make sure we can still finalize the transaction
+        fin_res = self.nodes[0].finalizepsbt(psbt)
+        assert_equal(fin_res["complete"], True)
+        fin_hex = fin_res["hex"]
+
+        # Change the sighash field to a different value and make sure we still finalize to the same thing
```

Comment:
What scenario is not tested if this flow is not done and the transaction generated above is mined directly?

Reply:
Nothing, it just helps to mutate one which we know is definitely good.